code/mission/import/xwingmissionparse.cpp

```diff
@@ -0,0 +1,743 @@
+		else
+			sprintf(sexp_buf, "( fotg-is-ship-identified \"%s\" )", arrival_fg_name);
+		Mp = sexp_buf;
+		return get_sexp_main();
+	}
+
+	return Locked_sexp_true;
+}
+
+int xwi_determine_anchor(const XWingMission *xwim, const XWMFlightGroup *fg)
+{
+	int mothership_number = fg->mothership;
+
+	if (mothership_number >= 0)
+	{
+		if (mothership_number < (int)xwim->flightgroups.size())
+			return get_parse_name_index(xwim->flightgroups[mothership_number].designation.c_str());
+		else
+			Warning(LOCATION, "Mothership number %d is out of range for Flight Group %s", mothership_number, fg->designation.c_str());
+	}
+
+	return -1;
+}
+
+const char *xwi_determine_formation(const XWMFlightGroup *fg)
+{
+	switch (fg->formation)
+	{
+		case XWMFormation::f_Vic:
+			return "Double Vic";
+		case XWMFormation::f_Finger_Four:
+			return "Finger Four";
+		case XWMFormation::f_Line_Astern:
+			return "Line Astern";
+		case XWMFormation::f_Line_Abreast:
+			return "Line Abreast";
+		case XWMFormation::f_Echelon_Right:
+			return "Echelon Right";
+		case XWMFormation::f_Echelon_Left:
+			return "Echelon Left";
+		case XWMFormation::f_Double_Astern:
+			return "Double Astern";
+		case XWMFormation::f_Diamond:
+			return "Diamond";
+		case XWMFormation::f_Stacked:
+			return "Stacked";
+		case XWMFormation::f_Spread:
+			return "Spread";
+		case XWMFormation::f_Hi_Lo:
+			return "Hi-Lo";
+		case XWMFormation::f_Spiral:
+			return "Spiral";
+	}
+
+	return nullptr;
+}
+
+const char *xwi_determine_base_ship_class(const XWMFlightGroup *fg)
+{
+	switch (fg->flightGroupType)
+	{
+		case XWMFlightGroupType::fg_X_Wing:
+			return "T-65c X-wing";
+		case XWMFlightGroupType::fg_Y_Wing:
+			return "BTL-A4 Y-wing";
+		case XWMFlightGroupType::fg_A_Wing:
+			return "RZ-1 A-wing";
+		case XWMFlightGroupType::fg_TIE_Fighter:
+			return "TIE/ln Fighter";
+		case XWMFlightGroupType::fg_TIE_Interceptor:
+			return "TIE/In Interceptor";
+		case XWMFlightGroupType::fg_TIE_Bomber:
+			return "TIE/sa Bomber";
+		case XWMFlightGroupType::fg_Gunboat:
+			return nullptr;
+		case XWMFlightGroupType::fg_Transport:
+			return "DX-9 Stormtrooper Transport";
+		case XWMFlightGroupType::fg_Shuttle:
+			return "Lambda-class T-4a Shuttle";
+		case XWMFlightGroupType::fg_Tug:
+			return nullptr;
+		case XWMFlightGroupType::fg_Container:
+			return nullptr;
+		case XWMFlightGroupType::fg_Freighter:
+			return "BFF-1 Bulk Freighter";
+		case XWMFlightGroupType::fg_Calamari_Cruiser:
+			return "Liberty Type Star Cruiser";
+		case XWMFlightGroupType::fg_Nebulon_B_Frigate:
+			return "EF76 Nebulon-B Escort Frigate";
+		case XWMFlightGroupType::fg_Corellian_Corvette:
+			return "CR90 Corvette";
+		case XWMFlightGroupType::fg_Imperial_Star_Destroyer:
+			return "Imperial Star Destroyer";
+		case XWMFlightGroupType::fg_TIE_Advanced:
+			return nullptr;
+		case XWMFlightGroupType::fg_B_Wing:
+			return "B-wing Starfighter";
+		default:
+			break;
+	}
```

Comment:
No. No. Absolutely not. This is unacceptable. This not only _should_ be something defined in a table somewhere, it _must_ be. 

Yes, yes, yes, I do realize that this is highly purpose-bound code and all that, but again: We should not encode this strong a dependency on anything that isn't FS2 retail.